go/libraries/doltcore/sqle/diff_table.go

```diff
@@ -0,0 +1,324 @@
+		map[string]rowconv.ColNamingFunc{
+			diff.To:   toNamer,
+			diff.From: fromNamer,
+		})
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	root1, err := dEnv.WorkingRoot(ctx)
+
+	if err != nil {
+		panic(err)
+	}
+
+	root2, err := dEnv.StagedRoot(ctx)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return &DiffTable{name, dEnv, sch, j, root2, root1, "current", "HEAD"}
+}
+
+func (dt *DiffTable) Name() string {
+	return DoltDiffTablePrefix + dt.name
+}
+
+func (dt *DiffTable) String() string {
+	return DoltDiffTablePrefix + dt.name
+}
+
+func (dt *DiffTable) Schema() sql.Schema {
+	sqlSch, err := doltSchemaToSqlSchema(dt.Name(), dt.joiner.GetSchema())
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	return sqlSch
+}
+
+func toNamer(name string) string {
+	return diff.To + "_" + name
+}
+
+func fromNamer(name string) string {
+	return diff.From + "_" + name
+}
+
+func (dt *DiffTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
+	return &doltTablePartitionIter{}, nil
+}
+
+func tableData(ctx *sql.Context, root *doltdb.RootValue, tblName string, ddb *doltdb.DoltDB) (types.Map, schema.Schema, error) {
+	tbl, ok, err := root.GetTable(ctx, tblName)
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	var data types.Map
+	if !ok {
+		data, err = types.NewMap(ctx, ddb.ValueReadWriter())
+	} else {
+		data, err = tbl.GetRowData(ctx)
+	}
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	sch, err := tbl.GetSchema(ctx)
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	return data, sch, nil
+}
+
+func (dt *DiffTable) PartitionRows(ctx *sql.Context, part sql.Partition) (sql.RowIter, error) {
+	fromData, fromSch, err := tableData(ctx, dt.fromRoot, dt.name, dt.dEnv.DoltDB)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toData, toSch, err := tableData(ctx, dt.toRoot, dt.name, dt.dEnv.DoltDB)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromConv, err := rowconv.RowConvForSuperSchema(fromSch, dt.superSch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toConv, err := rowconv.RowConvForSuperSchema(toSch, dt.superSch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sch := dt.joiner.GetSchema()
+	toCol, ok := sch.GetAllCols().GetByName(toCommit)
+
+	if !ok {
+		panic("missing required column")
+	}
+
+	fromCol, ok := sch.GetAllCols().GetByName(fromCommit)
+
+	if !ok {
+		panic("missing required column")
+	}
+
+	return NewDiffRowItr(dt.joiner, fromData, toData, fromConv, toConv, dt.fromCommitVal, dt.toCommitVal, fromCol.Tag, toCol.Tag), nil
+}
+
+var _ sql.RowIter = (*DiffRowItr)(nil)
+
+type DiffRowItr struct {
+	ad      *diff.AsyncDiffer
+	diffSrc *diff.RowDiffSource
+	sch     schema.Schema
+	to      string
+	from    string
+	fromTag uint64
+	toTag   uint64
+}
+
+func NewDiffRowItr(joiner *rowconv.Joiner, rowDataFrom, rowDataTo types.Map, convFrom, convTo *rowconv.RowConverter, from, to string, fromTag, toTag uint64) *DiffRowItr {
```

Comment:
Probably should be unexported

Reply:
fromHash, toHash

Reply:
this isn't necessarily a hash.  could be a branch name.